docs/integrations/Instructions.md

```diff
@@ -0,0 +1,160 @@
+# Instructions
+
+Instructions have two forms, In and Out. For a transaction originating on a
+connected network, an In Instruction must be provided, which may embed an Out
+Instruction. For a transaction originating on Serai, only an Out Instruction is
+allowed. Additionally, short hand forms are provided to minimize representations
+on connected networks.
+
+Instructions are interpreted according to their non-Serai network. Addresses
+have no validation performed, beyond being a valid enum entry (when applicable)
+of the correct length, unless otherwise noted. If the processor is instructed to
+act on invalid data, or send to itself, it will drop the entire instruction.
+
+### Serialization
+
+  - Numbers are exclusively unsigned and encoded as compact integers under
+SCALE. If omitted, `0`.
+  - Enums are prefixed by an ordinal byte of their type, followed by their
+actual values.
+  - Vectors are prefixed by their length. If omitted, `vec![]`.
+  - Instruction fields are numbered and sequentially encoded, each prefixed by
+an ordinal byte. All other fields are sequentially encoded with no markers.
+
+Certain fields may be omitted depending on the network in question.
+
+### In Instructions
+
+  - `origin` (Address): Address from the network of origin which sent funds in.
+  - `target` (Address): The ink! contract to transfer the incoming funds to.
+  - `data`   (Vec<u8>): The data to call the target with.
+
+Networks may automatically provide `origin`. If they do, the instruction may
+still provide `origin`, overriding the automatically provided value. If no
+`origin` is provided, the instruction is dropped.
+
+Upon receiving funds, the respective Serai Asset contract is called, minting the
+appropriate amount of coins, and transferring them to the specified target with
+the attached data.
+
+If the transaction fails, funds are scheduled to be returned to `origin`.
+
+### Out Instructions
+
+  - `destination` (Enum { Native(Address), Serai(Address) }): Address to receive
+funds to.
```

Comment:
To elaborate, making any bridge a first class citizen doesn't really save on complexity, especially due to the scope creep. It would decrease gas costs on ETH, yet with increased first-party complexity when those are already a large burden, and this keeps thing modular. There is another discussion a bridge still needs to be a first class citizen to enable redemptions (/ we need a new API exposed for that), yet we're not tackling any bridge discussions yet (not even as a tracking issue, and I'd prefer to keep it that way). We're solely acknowledging API flexibility.